internal/v5/archive_test.go

```diff
@@ -124,6 +141,21 @@ func (s *ArchiveSuite) TestGetDevelopment(c *gc.C) {
 	})
 }
 
+func (s *commonSuite) assertArchiveDownload(c *gc.C, url, id string, archiveBytes []byte) {
```

Comment:
Given you have this function now, I think it should be easy to check
the archive contents always (and perhaps even do the range check).
How about:

     // assertArchiveDownload asserts that the given charm archive can be
     // downloaded from the path with the given id, and that the resulting
     // EntityIdHeader is equal to expectId. If expectId is empty, id will be
     // used instead.
     assertArchiveDownload(c, id string, archiveArchive *charm.CharmArchive, expectId string)

We can append the "/archive" suffix to id within the function.